### DISCUSSION TITLE
TimeOut issue

### DISCUSSION THREAD

**Discussion by KalaiJay** [mockoon/mockoon#961] [Help] [answered]
Hi,

When I try to make a asynchronous call to Mockoon from my Asp.netCore Api, I need to set the time out as minimum of 3 sec in my request, but my SLA is for 1.5 seconds. But if I set 1.5 sec I am getting timeout error.  This is how I call Mockoon from my Api

await client.PostAsync(EndPoint, requestContent, Token)

EndPoint = MocckApi, requestContent is the request object and token has the timeout value. Is there any way to improve the request response time for Mockoon calls.

**Comment by 255kb** [comment, marked as answer]
Mockoon response time is usually in the order of dozens/hundreds of milliseconds, normally not seconds. 
Did you add a latency to your mock API or your route? 
![image](https://user-images.githubusercontent.com/7489814/221126952-25d1cf37-0cb1-4e38-a3f2-4f073a848a0f.png)
![image](https://user-images.githubusercontent.com/7489814/221127021-d0b0bc36-abf7-4614-9a74-121290c3a3ac.png)

**Reply by KalaiJay** [reply]
Thanks for the reply. I have already set the Latency to 0. Still it takes time to receive the response. Is there any other way we can reduce the response time? FYI, I have created a docker image of Mockoon files and hosted the image in GCP and testing it from there.

**Reply by 255kb** [reply]
The response time is due to a number of parameters like your internet connection, the GCP region/data center you choose, the instance CPU and RAM, etc.
So, it's very hard to tell what you could improve.
Example: if you are using a small free GCP instance in the US and are making calls from the other side of the planet, having a >1s response time is not unusual.

Inside mockoon two main things could slow down a response: generating tons of fake data, adding some latency. Apart from that, it should answer quickly enough (~100ms).

**Reply by KalaiJay** [reply]
ok.. thanks of your response.

**Reply by KalaiJay** [reply]
Hi,

In my application I am setting timeout to 1500 ms, where as in Mockoon I am setting latency to 500 ms. I am getting timeout. Whereas, if I set the latency in mockoon to 0 it works fine. But in one of our testcase we need to set latency and test. Can you please help me out on this  issue.

**Reply by 255kb** [reply]
The same remarks as before apply. If you add a 500ms latency Mockoon should respond after a bit more than 500ms (500 + the time to process the request). Then, you add the time for the response to reach your app, and you get the total time. If it takes 1s to reach your app you may exceed the 1500ms timeout. 
If it's for a test, I suggest that you increase your timeout.
Aside from that, there is nothing we can do on our side. If you find a reproducible bug increasing the latency we would be happy to investigate.

**Reply by KalaiJay** [reply]
Thanks for your response. I have another question related to the latency. If I set latency to 500ms  for 200 ok response, it is working as expected, whereas if I set the latency to 500ms and set 400 status code in response I am getting timeouts. Can you help me in fixing this issue.

**Reply by 255kb** [reply]
Sorry, but I can't help without more information. Having a timeout is not an issue in itself, especially if your app has a timeout of 1.5s, it's very short.
If you can identify an issue in Mockoon and provide reproducible steps, I can investigate and help. But your timeout could come from 100 other reasons outside Mockoon, and right now without more information it's impossible to tell.

**Reply by KalaiJay** [reply]
Thanks for the reply. 
Here are the details. for 200 ok status code 

![Screenshot1](https://github.com/mockoon/mockoon/assets/108901937/3b64e24b-ad3e-459e-906e-3fd140074dff)

For 400 - bad request:

![Screenshot2](https://github.com/mockoon/mockoon/assets/108901937/091b8c85-b33f-4732-abd9-d2998992fdd2)

For 200 ok status code it is working fine, but 400 status code getting time out.
In Mockoon I have not configured any other settings

**Reply by 255kb** [reply]
Sorry, but you gave the exact same explanation as before. I understand that you get timeouts in your app when answering with 400, but there is no reason that this happens solely because of Mockoon. 
You can [share your full setup](https://mockoon.com/docs/latest/mockoon-data-files/environment-clipboard-copy/#copy-an-environment-or-a-route-json-to-the-clipboard) with me and I will have a look and test locally. If I cannot reproduce the timeout, it's likely due to a misconfiguration or network network issue independent of Mockoon.

**Reply by KalaiJay** [reply]
Thanks for your response. I will investigate from my end and will get back to you
